devel/fortio/fortiogrpc/grpcrunner.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fortiogrpc
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"runtime"
+	"runtime/pprof"
+
+	"istio.io/istio/devel/fortio"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/health/grpc_health_v1"
+)
+
+// TODO: refactor common parts between http and grpc runners
+
+// GrpcRunnerResults is the aggregated result of an GrpcRunner.
+// Also is the internal type used per thread/goroutine.
+type GrpcRunnerResults struct {
+	fortio.RunnerResults
+	client   grpc_health_v1.HealthClient
+	req      grpc_health_v1.HealthCheckRequest
+	RetCodes map[grpc_health_v1.HealthCheckResponse_ServingStatus]int64
+}
+
+// Used globally / in TestGrpc() TODO: change periodic.go to carry caller defined context
+var (
+	grpcstate []GrpcRunnerResults
+)
+
+// TestGrpc exercises Grpc health check at the target QPS.
+// To be set as the Function in RunnerOptions.
+func TestGrpc(t int) {
```

Comment:
yeah, on the other hand fortio is all about testing but I'll see if I can come up with something